analytics.js

```diff
@@ -1047,6 +1047,16 @@ class Analytics {
     return true;
   }
 
+  isDatasetAvailable() {
+    try {
+      const element = document.createElement("script");
```

Comment:
Polyfill has a better logic. Less code.
```
var t=document.createElement("div")
return t.setAttribute("data-a-b","c"),t.dataset&&"c"==t.dataset.aB;
```

Reply:
Getting syntax error using this syntax in IE 10
<img width="674" alt="Screenshot 2022-08-02 at 10 20 51 PM" src="https://user-images.githubusercontent.com/36885121/182430698-2f7a91ac-be68-4715-95ff-162f16cb9525.png">.

Reply:
Keeping the old one

Reply:
I'm unsure of the syntax error in the previous one, but there was a logical error. Fixed it.
```
var t=document.createElement("div")
return t.setAttribute("data-a-b","c"),t.dataset ? "c"===t.dataset.aB : false;
```
![image](https://user-images.githubusercontent.com/88789928/182434832-62db83b7-1053-4325-9598-7cb8daadd606.png)

Reply:
Now there is no error but getting true.
<img width="636" alt="Screenshot 2022-08-02 at 11 12 42 PM" src="https://user-images.githubusercontent.com/36885121/182439648-2405cf51-f867-48a2-b3c6-a4d0f74ea494.png">

Reply:
very wired

Reply:
Are you sure you are testing in IE 10?
Dev tools nav bar is looking a bit different than mine.

Reply:
I tried with a different browser. The result is the same.